tests/main/snapd-state/task.yaml

```diff
@@ -14,9 +14,19 @@ execute: |
     "$TESTSTOOLS"/snapd-state --foo 2>&1 | MATCH "snapd-state: no such command: --foo"
     "$TESTSTOOLS"/snapd-state foo 2>&1 | MATCH "snapd-state: no such command: foo"
 
+    function check_date() {
+        local TEST_DATE="$1"
+        local TOLERANCE_SECONDS="1800" # 30 minutes
+
+        CURRENT_DATE="$(date +'%F')"
+        TIMESTAMP_MINUS_TOLERANCE=$(($(date +%s) - TOLERANCE_SECONDS))
+        DATE_MINUS_TOLERANCE="$(date +'%F' --date="@${TIMESTAMP_MINUS_TOLERANCE}")"
+        echo "$TEST_DATE" | MATCH "^(${CURRENT_DATE}|${DATE_MINUS_TOLERANCE})"
```

Comment:
I wonder if we shouldn't change this MATCH-based check (that looks a bit fragile and also hard to understand) to integer comparison instead, i.e. abs of (current date - test timestamp) should be less than tolerance seconds, where dates are converted to seconds with "date +'%s'"?

Reply:
Makes sense indeed, thanks! Done :-)